screen2.0/src/app/_scatterPlot/scatterPlot.tsx

```diff
@@ -418,43 +432,54 @@ function Umap({ width: parentWidth, height: parentHeight, pointData: umapData, l
-                                        </React.Fragment>
-                                    ))}
-                                    <rect
-                                        width={parentWidth - 100}
-                                        height={parentHeight - 100}
-                                        fill="#0d0f98"
-                                        fillOpacity={0.2}
-                                        stroke="#0d0f98"
-                                        strokeWidth={4}
-                                        rx={8}
-                                        transform={zoom.toStringInvert()}
-                                    />
-                                    </g>
+                                        <svg width={(parentWidth-100) / 4} height={(parentHeight-100) / 4}>
+                                            <g
+                                            clipPath="url(#clip-minimap)"
+                                            transform={`
+                                                scale(0.25)
+                                            `}
+                                            >
+                                            <rect width={parentWidth - 100} height={parentHeight - 100} fill="white" stroke='grey' strokeWidth={4} rx={8}/>
+                                            {umapData.map((point, i) => (
+                                                <React.Fragment>
+                                                <circle
+                                                    cx={xScale(point.x)}
+                                                    cy={yScale(point.y)}
+                                                    r={3}
+                                                    fill={point.color}
+                                                />
+                                                </React.Fragment>
+                                            ))}
+                                            <rect
+                                                width={parentWidth - 100}
+                                                height={parentHeight - 100}
+                                                fill="#0d0f98"
+                                                fillOpacity={0.2}
+                                                stroke="#0d0f98"
+                                                strokeWidth={4}
+                                                rx={8}
+                                                transform={zoom.toStringInvert()}
+                                            />
+                                            </g>
+                                        </svg>
+                                    </div>,
+                                    miniMap.ref ? miniMap.ref.current : document.body
                                 )}
-                            </svg>
+                            
                             {useEffect(() => {
```

Comment:
"React Hook "useEffect" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function."

ESLint doesn't like this here. I dont think you're allowed to call hooks inside the body of return statement. Not sure where it should be moved though in this context. Seems to work though so idk?? Weird

Download the ESLint extension on vscode to get these warnings to pop up